model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/embedded/JGroupsConfigurator.java

```diff
@@ -126,6 +136,42 @@ public static void configureTopology(Config.Scope config, ConfigurationBuilderHo
         }
     }
 
+    static void createJGroupsProperties(ProviderConfigurationBuilder builder) {
+        Util.copyFromOption(builder, BIND_ADDRESS, "address", ProviderConfigProperty.STRING_TYPE, CachingOptions.CACHE_EMBEDDED_NETWORK_BIND_ADDRESS, false);
+        Util.copyFromOption(builder, BIND_PORT, "port", ProviderConfigProperty.INTEGER_TYPE, CachingOptions.CACHE_EMBEDDED_NETWORK_BIND_PORT, false);
+        Util.copyFromOption(builder, EXTERNAL_ADDRESS, "address", ProviderConfigProperty.STRING_TYPE, CachingOptions.CACHE_EMBEDDED_NETWORK_EXTERNAL_ADDRESS, false);
+        Util.copyFromOption(builder, EXTERNAL_PORT, "port", ProviderConfigProperty.INTEGER_TYPE, CachingOptions.CACHE_EMBEDDED_NETWORK_EXTERNAL_PORT, false);
+    }
+
+    private static void configureTransport(Config.Scope config) {
+        setJGroupsNetworkProperty(BIND_ADDRESS, () -> config.get(BIND_ADDRESS));
+        setJGroupsNetworkProperty(BIND_PORT, intAsString(config, BIND_PORT));
+        setJGroupsNetworkProperty(EXTERNAL_ADDRESS, () -> config.get(EXTERNAL_ADDRESS));
+        setJGroupsNetworkProperty(EXTERNAL_PORT, intAsString(config, EXTERNAL_PORT));
+    }
+
+    private static Supplier<String> intAsString(Config.Scope config, String key) {
+        Integer val = config.getInt(key);
+        return val == null ? () -> null : val::toString;
+    }
+
+    private static void setJGroupsNetworkProperty(String key, Supplier<String> config) {
+        String userConfig = config.get();
+        if (userConfig == null) {
+            // User property is either already set or missing, so do nothing
+            return;
+        }
+
+        String[] split = key.split("(?=[A-Z])", 3);
+        String property = "jgroups.%s.%s".formatted(split[1].toLowerCase(), split[2].toLowerCase());
```

Comment:
I assume this will work, but is hard to understand for me, and might fail with future changes. Could we have a switch statement instead which those values that we support? If you want to make it fancy, try using enums to make it safe.

Reply:
Good point, looking at this again I realise there's actually a bug as the external address properties are in the format `jgroups.%s_%s` not `jgroups.%s.%s`.

I have reworked to utilise an Enum.